File: app/src/main/java/com/qihuan/find/model/db/AppDatabase.java
package com.qihuan.find.model.db;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.qihuan.find.common.AppConfig;
import com.qihuan.find.model.bean.find.CollectionBean;
import com.qihuan.find.model.db.dao.CollectionDao;

/**
 * AppDatabase
 * Created by Qi on 2017/8/24.
 */
@Database(entities = {CollectionBean.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    private static volatile AppDatabase INSTANCE;

    public synchronized static AppDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context, AppDatabase.class, AppConfig.DATABASE_NAME).build();
                }
            }
        }
        return INSTANCE;
    }

    public abstract CollectionDao collectionDao();
}

File: ModuleCommon/src/main/java/com/qihuan/commonmodule/db/AppDatabase.kt
package com.qihuan.commonmodule.db

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import com.qihuan.commonmodule.base.BaseApp
import com.qihuan.commonmodule.collection.CollectionBean
import com.qihuan.commonmodule.collection.CollectionDao


/**
 * AppDatabase
 * @author qi
 * @date 2018/10/17
 */
@Database(entities = [CollectionBean::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun collectionDao(): CollectionDao

    companion object {
        private const val APP_DATABASE_NAME = "find.db"
        val instance: AppDatabase by lazy {
            Room.databaseBuilder(BaseApp.instance, AppDatabase::class.java, APP_DATABASE_NAME).build()
        }
    }
}